src/pdf.rs

```diff
@@ -0,0 +1,673 @@
+
+                // Update current_layer, current_y_pos, and current_page with the returned values
+                current_layer = new_layer;
+                current_y_pos = Mm(273.0) - line_height;
+                current_page = new_page;
+            }
+
+            let work_time_str = format_minutes_as_time(*work_time);
+            let ride_time_str = format_minutes_as_time(*ride_time);
+
+            // Calculate the total time (work + ride)
+            let total_time = work_time + ride_time;
+            let total_time_str = format_minutes_as_time(total_time);
+
+            let points = vec![
+                (Point::new(Mm(21.0), current_y_pos - Mm(3.0)), false),
+                (
+                    Point::new(Mm(21.0), current_y_pos + line_height - Mm(3.0)),
+                    false,
+                ),
+                (
+                    Point::new(Mm(188.0), current_y_pos + line_height - Mm(3.0)),
+                    false,
+                ),
+                (Point::new(Mm(188.0), current_y_pos - Mm(3.0)), false),
+            ];
+            let rectangle = Line {
+                points,
+                is_closed: true,
+                has_fill: true,
+                has_stroke: false,
+                is_clipping_path: false,
+            };
+            let header_colour = Color::Rgb(Rgb::new(0.952, 0.952, 0.952, None));
+            current_layer.set_fill_color(header_colour);
+            current_layer.add_shape(rectangle);
+
+            let text_colour = Color::Rgb(Rgb::new(0.0, 0.0, 0.0, None));
+            current_layer.set_fill_color(text_colour);
+
+            // Output task_id rows
+            current_layer.use_text("", font_size, column_widths[0], current_y_pos, &font_medium); // Empty first column
+            current_layer.use_text(
+                &work_time_str,
+                font_size,
+                column_widths[1],
+                current_y_pos,
+                &font_medium,
+            );
+            current_layer.use_text(
+                &ride_time_str,
+                font_size,
+                column_widths[2],
+                current_y_pos,
+                &font_medium,
+            );
+            current_layer.use_text(
+                &total_time_str,
+                font_size,
+                column_widths[3],
+                current_y_pos,
+                &font_medium,
+            );
+            current_layer.use_text(
+                &*task_id,
+                font_size,
+                column_widths[4],
+                current_y_pos,
+                &font_medium,
+            );
+
+            current_y_pos -= line_height;
+        }
+    }
+    let current_page_text = current_page.to_string();
+    current_layer.use_text(current_page_text, 13.0, Mm(185.0), Mm(14.0), &font_medium);
+
+    let pdf_name = format!("./target/Zeiterfassung {} {}.pdf", employee_id, given_month);
+    doc.save(&mut BufWriter::new(
+        File::create(pdf_name).unwrap(),
```

Comment:
Dont save the pdf file. return the pdf binary base64 encoded. Only save the file for debug purposes and thats done with #[cfg(debug_assertions)]